pkg/simulation/passenger/passenger_store.go

```diff
@@ -81,6 +84,126 @@ func (ps *PassengersStore) generatePassengers(c *city.City) {
 	}
 }
 
+func (ps *PassengersStore) GeneratePassengersDueModel(c *city.City, passengerModel []byte) error {
+	reader := csv.NewReader(bytes.NewReader(passengerModel))
+	reader.Comma = ','
+
+	records, err := reader.ReadAll()
+	if err != nil {
+		return fmt.Errorf("error reading passenger model csv: %w", err)
+	}
+
+	if len(records) == 0 {
+		return fmt.Errorf("passenger model csv is empty")
+	}
+
+	header := records[0]
+	if len(header) < 4 {
+		return fmt.Errorf("invalid header, expected at least 4 columns, got %d", len(header))
+	}
+
+	stopsByName := c.GetStopsByName()
+	resolveStopsID := func(stopName string) ([]uint64, error) {
+		if stopName == "" {
+			return nil, fmt.Errorf("empty stop group name")
+		}
+
+		if group, ok := stopsByName[stopName]; !ok || len(group) == 0 {
+			return nil, fmt.Errorf("stop group %q not found", stopName)
+		} else {
+			ids := make([]uint64, 0, len(group))
+			for id := range group {
+				ids = append(ids, id)
+			}
+			return ids, nil
+		}
+	}
+
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+
+	ps.passengersToSpawn = make(map[uint][]passengerSpawn)
+
+	var counter uint64
```

Comment:
I don't think we need the `counter`, as we can just use the `i` from the `for` loop below